File: lifecycle/src/main/java/org/sheedon/lifecycle/LifecycleManager.java
package org.sheedon.lifecycle;

import java.util.HashSet;
import java.util.Set;

/**
 * 生命周期管理类
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2021/12/20 3:39 下午
 */
public class LifecycleManager implements ILifecycle {

    private final Set<ILifecycle> elements = new HashSet<>();

    /**
     * 追加持有生命周期的元素
     *
     * @param element 持有生命周期的元素
     */
    public void addLifecycle(ILifecycle element) {
        if (element == null) {
            return;
        }

        elements.add(element);
    }

    @Override
    public void onDestroy() {
        for (ILifecycle element : elements) {
            if (element == null) continue;
            element.onDestroy();
        }

        elements.clear();
    }
}

File: lifecycle/src/main/java/org/sheedon/lifecycle/LifecycleManager.kt
package org.sheedon.lifecycle

/**
 * 生命周期管理类
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2022/1/5 11:37 上午
 */
class LifecycleManager : ILifecycle {

    private val elements = HashSet<ILifecycle>()

    /**
     * 追加持有生命周期的元素
     *
     * @param element 持有生命周期的元素
     */
    fun addLifecycle(element: ILifecycle) {
        elements.add(element)
    }


    @Override
    override fun onDestroy() {
        elements.forEach { it.onDestroy() }
        elements.clear()
    }
}